src/main/scala/com/ing/wbaa/rokku/sts/vault/VaultService.scala

```diff
@@ -0,0 +1,82 @@
+package com.ing.wbaa.rokku.sts.vault
+
+import java.nio.charset.StandardCharsets
+
+import akka.actor.ActorSystem
+import com.bettercloud.vault.response.VaultResponse
+import com.bettercloud.vault.{Vault, VaultConfig}
+import com.ing.wbaa.rokku.sts.config.VaultSettings
+import com.ing.wbaa.rokku.sts.data.UserName
+import com.ing.wbaa.rokku.sts.data.aws.AwsCredential
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.jdk.CollectionConverters._
+import scala.util.{Failure, Success, Try}
+
+trait VaultService extends LazyLogging {
+
+  protected[this] def vaultSettings: VaultSettings
+
+  protected[this] implicit def system: ActorSystem
+
+  protected[this] implicit lazy val vaultExecutionContext: ExecutionContext =
+    Try {
+      system.dispatchers.lookup("vault-dispatcher")
+    } match {
+      case Success(dispatcher) => dispatcher
+      case Failure(ex) =>
+        logger.error("Failed to configure dedicated vault dispatcher, using default one, " + ex.getMessage)
+        system.dispatcher
+    }
+
+  protected lazy val vault: Vault = {
+    val vault = new Vault(new VaultConfig()
+      .address(vaultSettings.vaultUrl)
+      .engineVersion(2)
+      .token(vaultSettings.vaultToken)
+      .readTimeout(vaultSettings.readTimeout)
+      .openTimeout(vaultSettings.openTimeout)
+      .build())
+    vault
+  }
+
+  def insertNpaCredentialsToVault(username: UserName, awsCredential: AwsCredential): Future[Boolean] = Future {
+    val secretsToSave: Map[String, AnyRef] = Map("accessKey" -> awsCredential.accessKey.value, "secretKey" -> awsCredential.secretKey.value)
```

Comment:
what is the reason for AnyRef for secretKey? I think it should be at least string as well?

Reply:
Vault client dependency method signature for writing is as follows:
`public LogicalResponse write(final String path, final Map<String, Object> nameValuePairs)` and with no overloads which provide a different String generic.

The vaule is being used here:

`.write(vaultSettings.vaultPath + "/" + username.value, secretsToSave.asJava)`

However `insertNpaCredentialsToVault` takes `UserName` and `AwsCredential` objects as parameters, hence I don't see this as a potential problem.